Sources/Purchasing/OfferingsManager.swift

```diff
@@ -111,25 +127,53 @@ class OfferingsManager {
 
 private extension OfferingsManager {
 
-    func handleOfferingsBackendResult(
-        with response: OfferingsResponse,
+    func fetchCachedOfferingsFromDisk(
+        appUserID: String,
         fetchPolicy: FetchPolicy,
-        completion: (@MainActor @Sendable (Result<Offerings, Error>) -> Void)?
+        completion: (@escaping @Sendable (Offerings?) -> Void)
+    ) {
+        guard let data = self.deviceCache.cachedOfferingsData(appUserID: appUserID),
+              let response: OfferingsResponse = try? JSONDecoder.default.decode(jsonData: data) else {
+            completion(nil)
+            return
+        }
+
+        self.createOfferings(
+            from: response,
+            fetchPolicy: fetchPolicy,
+            completion: { [cache = self.deviceCache] result in
+                switch result {
+                case let .success(offerings):
+                    Logger.debug(Strings.offering.vending_offerings_cache_from_disk)
+
+                    cache.cacheInMemory(offerings: offerings)
```

Comment:
Important to cache in memory in this case too I think. Covered by a test.

Reply:
This makes sense but we should keep the offerings in memory cache as "stale" in this case so we retry getting the data from the backend as soon as possible.

Reply:
I still think we might want to keep the in memory offerings as "stale" in this case, so we retry fetching as soon as possible. Currently this will store the offerings in memory and keep them there for the full duration of the cache. Thoughts?

Reply:
Oh great point!

Reply:
26b79868841546137d0da5ad897507530ddbe1d8